File: Frontend/lvtn_app/lib/api/sgddt_api.dart
import 'package:dio/dio.dart';
import 'package:lvtn_app/api/api_client.dart';

class SGDDTApi {
  static Future<Response> getListThiSinh() async {
    try {
      final response = await ApiClient.dio.get('/sgddt/thisinh');
      return response;
    } catch (e) {
      rethrow;
    }
  }

  static Future<Response> getKQThiBySBD({required String sbd}) async {
    try {
      final response = await ApiClient.dio.get('/sgddt/thisinh/$sbd');
      return response;
    } catch (e) {
      rethrow;
    }
  }

  static Future<Response> updateKQThiBySBD({
    required String sbd,
    required String maHS,
    double? diemToan,
    double? diemVan,
    double? diemMonThu3,
    double? diemMonChuyen,
  }) async {
    try {
      FormData formData = FormData.fromMap({
        'maHS': maHS,
        'diemToan': diemToan,
        'diemVan': diemVan,
        'diemMonThu3': diemMonThu3,
        'diemMonChuyen': diemMonChuyen,
      });
      final response = await ApiClient.dio.patch(
        '/sgddt/thisinh/kqthi/$sbd',
        data: formData,
        options: Options(contentType: 'multipart/form-data'),
      );
      return response;
    } catch (e) {
      rethrow;
    }
  }

  static Future<Response> getTrangThai() async {
    try {
      final response = await ApiClient.dio.get('/sgddt/trangthai');
      return response;
    } catch (e) {
      rethrow;
    }
  }

  static Future<Response> updateTrangThai({
    required int maTT,
    required String kieuDuLieu,
    required String newVal,
  }) async {
    try {
      final response = await ApiClient.dio.patch(
        '/sgddt/trangthai/$maTT',
        queryParameters: {'kieuDuLieu': kieuDuLieu, 'newVal': newVal},
      );
      return response;
    } catch (e) {
      rethrow;
    }
  }
}

File: Frontend/lvtn-web/src/api/sgddt/sgddtApi.js
import axiosClient from "../axiosClient";

const sgddtApi = {
  getTrangThai: () => {
    return axiosClient.get("/sgddt/trangthai");
  },
  getTrangThaiByMaTT: (maTT) => {
    return axiosClient.get(`/sgddt/trangthai/${maTT}`);
  },
  updateTrangThai: (maTT, kieuDuLieu, newVal) => {
    return axiosClient.patch(`/sgddt/trangthai/${maTT}`, null, {
      params: { kieuDuLieu, newVal },
    });
  },
  getThiSinh: () => {
    return axiosClient.get("/sgddt/thisinh");
  },
  findThiSinh: (soBaoDanh) => {
    return axiosClient.get(`/sgddt/thisinh/${soBaoDanh}`);
  },
  searchThiSinh: (keyword) => {
    return axiosClient.get("/sgddt/thisinh/search", {
      params: { keyword: keyword },
    });
  },
  updateSBDAndPhongThi: () => {
    return axiosClient.post("/sgddt/thisinh/capnhat");
  },
  xetTuyen: () => {
    return axiosClient.patch("/sgddt/xettuyen");
  },
  updateKetQuaThi: (soBaoDanh, ketQuaThi) => {
    return axiosClient.patch(`/sgddt/thisinh/kqthi/${soBaoDanh}`, ketQuaThi, {
      headers: { "Content-Type": "multipart/form-data" },
    });
  },
  importKetQuaThi: (file, onUploadProgress) => {
    const formData = new FormData();
    formData.append("file", file);
    return axiosClient.patch("/sgddt/thisinh/kqthi/import", formData, {
      headers: { "Content-Type": "multipart/form-data" },
      onUploadProgress: (event) => {
        if (onUploadProgress) onUploadProgress(event);
      },
    });
  },
  exportKetQuaThi: (axiosConfig) => {
    return axiosClient.get("/sgddt/thisinh/kqthi/export", {
      responseType: "blob",
      ...axiosConfig,
    });
  },
  exportKetQuaTuyenSinh: (axiosConfig) => {
    return axiosClient.get("/sgddt/kqts/export", {
      responseType: "blob",
      ...axiosConfig,
    });
  },
};

export default sgddtApi;
